diff --git a/gn/util/exe_path.cc b/gn/util/exe_path.cc
--- a/gn/util/exe_path.cc
+++ b/gn/util/exe_path.cc
@@ -16,5 +16,7 @@
 #include <sys/sysctl.h>
 #include <sys/types.h>
+#elif defined(OS_HAIKU)
+#include <image.h>
 #endif
 
 #if defined(OS_MACOSX)
@@ -62,6 +64,20 @@ base::FilePath GetExePath() {
   return base::FilePath(buf);
 }
 
+#elif defined(OS_HAIKU)
+
+base::FilePath GetExePath() {
+  int32 cookie = 0;
+  image_info info;
+  while (get_next_image_info(B_CURRENT_TEAM, &cookie, &info) == B_OK) {
+    if (info.type == B_APP_IMAGE)
+      return base::FilePath(info.name);
+  }
+
+  NOTREACHED() << "Unable to resolve the Haiku application image path.";
+  return base::FilePath();
+}
+
 #else
 
 base::FilePath GetExePath() {
