diff --git a/chromium/skia/BUILD.gn b/chromium/skia/BUILD.gn
--- a/chromium/skia/BUILD.gn
+++ b/chromium/skia/BUILD.gn
@@ -473,7 +473,7 @@ component("skia") {
     ]
   }
 
-  if (is_linux || is_chromeos || is_android) {
+  if ((is_linux && !is_haiku) || is_chromeos || is_android) {
     sources += [
       # Retain the files for the SkFontMgr_Android on linux to emulate android
       # fonts. See content/zygote/zygote_main_linux.cc
@@ -551,10 +551,12 @@ component("skia") {
 
   if (is_linux || is_chromeos) {
     deps += [
-      "//third_party/expat",
       "//third_party/fontconfig",
       "//third_party/icu:icuuc",
     ]
+    if (!is_haiku) {
+      deps += [ "//third_party/expat" ]
+    }
   }
 
   if (is_android) {
